code/game/machinery/doors/door.dm

```diff
@@ -511,3 +508,64 @@
 	if(caller.checkpass(PASSGLASS))
 		return !opacity
 	return ..()
+
+/**
+ * Checks which way the airlock is facing and adjusts the direction accordingly.
+ * For use with multi-tile airlocks.
+ */
+/obj/machinery/door/proc/get_adjusted_dir(dir)
+	if(dir in list(EAST, WEST))
+		return EAST
+	else
+		return NORTH
+
+/**
+ * Sets the bounds of the airlock. For use with multi-tile airlocks.
+ * If the airlock is multi-tile, it will set the bounds to be the size of the airlock.
+ * If the airlock doesn't already have fillers, it will create them.
+ * If the airlock already has fillers, it will move them to the correct location.
+ */
+/obj/machinery/door/proc/update_bounds()
```

Comment:
If the door somehow moves this is going to cause issues as we don't reset the contents of `fillers` before adding to it. As such, please do a loop to delete all the contents of the `fillers` list and remove them from the list.